### PR TITLE
Clean up implementation following cgridDEV merge

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Clean up implementation following cgridDEV merge, based on comments in https://github.com/CICE-Consortium/CICE/pull/715
- [X] Developer(s): 
    apcraig, eclare108213
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    All bit-for-bit and still working as before, Tested on Onyx with 3 compilers, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#078aab4844983ccbb9c9fddc1a70e060e1d99a34
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

- Refactor capping namelist, added capping_method variable string for namelist,
  options are "hibler" or "kreyscher" equivalent to 1 and 0 in old
  capping namelist.  capping variable still exists and is set
  depending on capping_method setting.  Update ice_in and other set_nml options.
- Add box2001 option for ice_data_conc.  Does same things at p5, but makes
  decreases confusion in box2001 usage.  Updated set_nml options.
- Remove support for ice_data_type = smallblock, bigblock, easthalf.  Remove
  a couple tests that were using these.
- Capitalize last letter in variables dxt, dyt, dxe, dye, dxn, dyn, dxu, dyu ->
  dxT, dyT, dxE, dyE, dxN, dyN, dxU, dyU.  This is the general direction we
  want to go, but this was done to make "dyn" look less like "dynamics".
- Modify some comment indentation in ice_init.F90
- Reorder grid_average_X2Y calls in ice_dyn_evp.F90 for readability
- Remove older code in ice_dyn_shared.F90 referencing dte, dte2T, tdamp2, etc.
- Clean up comments in ice_history_shared.F90, ice_dyn_evp.F90, ice_transport_driver.F90,
  ice_flux.F90, ice_forcing.F90, ice_grid.F90
- Remove commented out code in ice_dyn_shared.F90 around computation of iceumask
- Remove commented out code in ice_forcing.F90 related to box2001_data_ocn
- Update documentation.
  - capping_method
  - namelist table was reviewed and updated for consistency with code
  - ice_ic change from default to internal
  - dxt -> dxT, etc

### CODE REVIEW

File: doc/source/science_guide/sg_dynamics.rst
Comment (eclare108213) [inline doc/source/science_guide/sg_dynamics.rst:508]: Can we use the labels 'max' or 'sum' for capping_method instead of people's names?  It's more intuitive, and I've always tried to avoid naming things after individual people except when it's unavoidable.

**Review by eclare108213 — CHANGES_REQUESTED** [review]
Looks great.  I only have one change request, which unfortunately does impact several files.  I hope that you wrote a script to make all of those capitalization changes at once!  Thank you.

File: doc/source/science_guide/sg_dynamics.rst
Comment (apcraig) [reply doc/source/science_guide/sg_dynamics.rst:508]: Happy to make that change, not a problem.  What strings do we want to use instead?  @JFLemieux73 @eclare108213  Just let me know and I'll make the change.

Comment (eclare108213) [reply doc/source/science_guide/sg_dynamics.rst:508]: I suggest 'max' instead of Hibler and 'sum' instead of Kreyscher

Comment (apcraig) [reply doc/source/science_guide/sg_dynamics.rst]: I changed the implementation to "max" and "sum".  I also added some tests (alt07) to test capping="sum" and visc_method="avg_zeta" and a different ndte.  We can continue to update alt07 as we work towards better coverage of the dynamics options.

**Comment by apcraig** [comment]
Thanks @DeniseWorthen, it sounds like you and @dabail10 are not synced up beyond maybe the cap.  Whatever answer changes you're seeing must be coming from from C-grid and/or cap issues.

It might be worthwhile for @dabail10 to update CICE to before the C-grid merge and verify results, cap should not have to change.  Then update to C-grid merge and update cap.

CICE standalone should be bit-for-bit with the C-grid merge.  But, the coupled system may not be.  Some of the averaging was moved out of the forcing/coupling code and into the dynamics.  This was done in a bit-for-bit way for standalone, but if the averaging was being done slightly differently in the cap than in the standlone forcing, this could result in answer changes.

I think this testing needs to be done carefully.  Let me know if you want me to help.  I'm willing to jump into CESM a bit and contribute.

**Comment by apcraig** [comment]
@eclare108213, just waiting for your review approval.  thanks.

**Review by eclare108213 — APPROVED** [review]

**Pull request merged**
